docs/DeveloperGuide.md

```diff
@@ -133,6 +133,30 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### List Students' Email Feature
+The list students' email feature allows the end-user to retrieve a list of students' emails, which are concatenated with
+a semi-colon `;`. This allows for easy copy and pasting to e-mail applications, such as Microsoft Outlook, for mass
+e-mail purposes (e.g. newsletter). 
+
+This feature is mainly supported by `EmailCommand`, with retrieval of students' emails through the Model interface
+`Model#getFilteredStudentList()`.
+
+Below is an example of how the list students' email mechanism works:
+1. The user executes the list students' emails command with the command `emails`
+2. `LogicManager` receives the command, and hands off the parsing of command to `AddressBookParser`
+3. `AddressBookParser` recognises the command and creates a new `EmailCommand`
+4. `EmailCommand` is returned back to `LogicManager`, which then executes the command through `EmailCommand#execute()`
+5. Upon `EmailCommand#execute()`, a list of `Student` are retrieved through `Model#getFilteredStudentList()`
+6. The list of `Student` emails are then concatenated with ';' into a `String`
+7. The concatenated `String` is then returned to `LogicManager` as a new `CommandResult`
+8. The `CommandResult` containing the concatenated email is then displayed to the user through `ResultDisplay`
+
+The following activity diagram summarizes what happens when a user executes the `emails` command:
+![EmailCommandActivityDiagram.png](images/sam/EmailCommandActivityDiagram.png)
+
+The following sequence diagram summarizes what happens when a user executes the `emails` command:
+![EmailCommandSequenceDiagram.png](images/sam/EmailCommandSequenceDiagram.png)
```

Comment:
Hmmm I think line 154 to 155 and line 157 to 158 are duplicates ah.

Reply:
It's activity and sequence diagram! hahaha

Reply:
O.o damnnn u did both. Haha ok sorry. Let me go approve ah